### ISSUE TITLE
screen reader problems while reading code word by word

### ISSUE THREAD

**Issue by snufas** [Microsoft/vscode#34319]
- VSCode Version: Code 1.16.0 (787b31c0474e6165390b5a5989c9619e3e16f953, 2017-09-06T16:27:49.891Z)
- OS Version: Windows_NT x64 10.0.15063
- Extensions: none
---

Steps to Reproduce:
use NVDA screen reader with VS code editor it does not matter what language or file type.
CTRL+ right arrow to read word by word 
issues using screen reader. Screen reader repeats words multiple times when reading word by word when using VS code. attach in short recording of the problem and comparing with notepad++, the nvda screen reader interacts with the text differently while using VS studio. 
it would be nice if you watch the video and try to make sense what i am trying to explain. it would be very good if devs of VS code fix that. this fix would help blind or low vision community who uses screen reader to use VS code.
i hope that can be fixed because this is the problem that there is no way to get around and is very annoying to work with. 
thanks
https://w ww.dropbox.com/s/ujbemn92u90ojfe/nvda_vs_Code_problems.mp4? dl=0

**Comment by alexandrudima** [comment]
I think the problem with word navigation stems from the fact that we have a very peculiar type of word navigation (we stop both at the stand and at the end of words). I believe NVDA expects our word navigation to behave more like Notepad or Visual Studio and thus misreads the word or symbols that were jumped over.

We can look into adding some very specific word navigation commands that follow precisely what Notepad or Visual Studio do and enable those when NVDA is connected to VS Code.

**Comment by snufas** [comment]
i don't think that is NVDA fault, because it works fine in other editors.  i know that is a different story in terms of the notepad ++ or other windows editors. this editor is build using electron however i tried atom editor the text area didn’t worked at all. one thing i noticed that after a while of not using the editor and leaving in the background for a while, NVDA does not read the editor code even after focus using spacebar i found the way for this by using ALT F and reading a bit of menu in there and then focus back to read as it was before.  my understanding it is a problem that electron has somewhere down the line or NVDA gets some extra info it does not need the reading mechanism for NVDA worked same no matter the application if I can read the text the word by word navigation works as expected.

**Comment by Neurrone** [comment]
The inability to gain focus is tracked in #28316. This issue is a duplicate of #28306

**Comment by alexandrudima** [comment]
Let's track in #28316 and #28306
